### PR TITLE
SPCTR-1008 - New: Responsive Border Components 

### PR DESCRIPTION
**Pull request by imrantushar**
### Description
Added Border Responsive Components, used this component inside various block

### Types of changes
New Features

### How has this been tested?
Goto Heading Block Highlight style settings => border panel

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: src/blocks/wp-search/attributes.js [hunk 1]
```diff
@@ -21,6 +25,23 @@ const attributes = {
 			styleType: 'input-size'
 		}
 	},
+	borderStyle: {
+		type: 'string',
+		default: 'none',
+	},
```
Comment (supriyajjadhav) [inline src/blocks/wp-search/attributes.js:31]: Please add copy-paste object

File: blocks-config/post/class-uagb-post.php [hunk 2]
```diff
@@ -487,21 +472,9 @@ public function get_post_attributes() {
...
-				'borderStyle'                   => array(
-					'type'    => 'string',
-					'default' => 'none',
-				),
-				'borderColor'                   => array(
-					'type'    => 'string',
-					'default' => '',
-				),
-				'borderHColor'                  => array(
+				'btnBorderHColor'               => array(
```
Comment (imrantushar) [inline blocks-config/post/class-uagb-post.php:475]: @supriyajjadhav please remove **btnBorderHColor** attribute, our function automatically generating it.

File: classes/class-uagb-block-helper.php [hunk 3]
```diff
@@ -33,6 +33,17 @@ public static function get_buttons_child_selectors( $attr, $id, $child_migrate )
 			$t_selectors = array();
 			$selectors   = array();
 
+			$borderCSS       = self::uag_generate_border_css( $attr, 'btn' );
```
Comment (RajkiranBagal) [inline classes/class-uagb-block-helper.php:36]: PHP variables should be `snake_case`

File: blocks-config/post/class-uagb-post.php [hunk 4]
```diff
@@ -69,6 +69,8 @@ public function register_blocks() {
 				return;
 			}
 
+			$paginationMasonryBorderAttribute = UAGB_Block_Helper::uag_generate_border_attribute( 'paginationMasonry' );
```
Comment (RajkiranBagal) [inline blocks-config/post/class-uagb-post.php:72]: PHP variables should be `snake_case`

File: blocks-config/post/class-uagb-post.php [hunk 5]
```diff
@@ -367,6 +355,7 @@ public function register_blocks() {
 		 * @since 0.0.1
 		 */
 		public function get_post_attributes() {
+			$btnBorderAttribute = UAGB_Block_Helper::uag_generate_border_attribute( 'btn' );
```
Comment (RajkiranBagal) [inline blocks-config/post/class-uagb-post.php:358]: PHP variables should be `snake_case`

File: classes/class-uagb-block-helper.php [hunk 6]
```diff
@@ -1096,6 +1103,164 @@ public static function uag_get_background_obj( $bg_obj ) {
 			return $gen_bg_css;
 		}
 
+		/**
+		 * Border attribute generation Function.
+		 *
+		 * @param  array $prefix   Attribute Prefix.
+		 *
```
Comment (RajkiranBagal) [inline classes/class-uagb-block-helper.php:1110]: Add since x.x.x tag for this function description

File: classes/class-uagb-block-helper.php [hunk 7]
```diff
@@ -1096,6 +1103,164 @@ public static function uag_get_background_obj( $bg_obj ) {
...
+			$border_attr[ "{$prefix}BorderColor" ]  = $defaults['borderColor'];
+			$border_attr[ "{$prefix}BorderHColor" ] = $defaults['borderHColor'];
+			return $border_attr;
+		}
+
+		/**
+		 * Border CSS generation Function.
+		 *
+		 * @param  array  $attr   Attribute List.
+		 * @param  string $prefix Attribuate prefix .
```
Comment (RajkiranBagal) [inline classes/class-uagb-block-helper.php:1182]: Add since x.x.x tag

File: classes/class-uagb-block-helper.php [hunk 8]
```diff
@@ -1096,6 +1103,164 @@ public static function uag_get_background_obj( $bg_obj ) {
...
+			}
+			return $gen_border_css;
+		}
+
+		/**
+		 * Deprecated Border CSS generation Function.
+		 *
+		 * @param  array  $current_css   Current style list.
+		 * @param  string $border_width   Border Width.
+		 * @param  string $border_radius Border Radius.
```
Comment (RajkiranBagal) [inline classes/class-uagb-block-helper.php:1234]: Add since x.x.x tag

File: includes/blocks/advanced-heading/attributes.php [hunk 9]
```diff
@@ -7,150 +7,149 @@
...
-	'highLightBottomPaddingTablet' => '',
-	'highLightTopPaddingMobile'    => '',
-	'highLightRightPaddingMobile'  => '',
-	'highLightLeftPaddingMobile'   => '',
-	'highLightBottomPaddingMobile' => '',
-	'highLightPaddingUnit'         => 'px',
-	'highLightPaddingUnitTablet'   => 'px',
-	'highLightPaddingUnitMobile'   => 'px',
-	'highLightPaddingLink'         => '',
+$highLightBorderAttribute = UAGB_Block_Helper::uag_generate_border_attribute( 'highLight' );
```
Comment (RajkiranBagal) [inline includes/blocks/advanced-heading/attributes.php:10]: PHP variables should be `snake_case` Do this change in all file where needed

**Review by supriyajjadhav — APPROVED** [review]

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: includes/blocks/container/frontend.css.php [hunk 10]
```diff
@@ -39,7 +39,17 @@
 
 $video_bg_css = UAGB_Block_Helper::uag_get_background_obj( $bg_obj_desktop );
 
-// Spacing Variables.
+
+$borderCSS       = UAGB_Block_Helper::uag_generate_border_css( $attr, 'container' );
+$borderCSS       = UAGB_Block_Helper::uag_generate_deprecated_border_css(
+	$borderCSS,
+	( isset( $attr['borderWidth'] ) ? $attr['borderWidth'] : '' ),
+	( isset( $attr['borderRadius'] ) ? $attr['borderRadius'] : '' ),
```
Comment (imrantushar) [inline includes/blocks/container/frontend.css.php:47]: @supriyajjadhav Don't needed extra array isset checking

File: includes/blocks/info-box/attributes.php [hunk 11]
```diff
@@ -211,9 +208,9 @@
 	'prefixMarginLeftMobile'     => '',
 	'prefixMobileMarginUnit'     => 'px',
 	'prefixTabletMarginUnit'     => 'px',
+	...$ctaBorderAttribute,
```
Comment (imrantushar) [inline includes/blocks/info-box/attributes.php:211]: @supriyajjadhav I think PHP old version will be not supported this operator, please cross check it or use array_merge.

File: includes/blocks/post-masonry/attributes.php [hunk 12]
```diff
@@ -98,11 +101,7 @@
 	'overlayOpacity'                    => '50',
 	'bgOverlayColor'                    => '#ffffff',
 	'ctaText'                           => __( 'Read More', 'ultimate-addons-for-gutenberg' ),
-	'borderWidth'                       => '',
-	'borderStyle'                       => 'none',
-	'borderColor'                       => '',
-	'borderHColor'                      => '',
-	'borderRadius'                      => '',
+	'btnBorderHColor'                   => '',
```
Comment (imrantushar) [inline includes/blocks/post-masonry/attributes.php:104]: @supriyajjadhav Please remove this attribute, our function automatically generating it.

File: includes/blocks/testimonial/attributes.php [hunk 13]
```diff
@@ -96,7 +94,7 @@
 	'columnGapType'           => 'px',
 	'descSpaceType'           => 'px',
 	'nameSpaceType'           => 'px',
-	'borderHoverColor'        => '',
+	'btnBorderHoverColor'     => '',
```
Comment (imrantushar) [inline includes/blocks/testimonial/attributes.php:97]: @supriyajjadhav it is wrong attribute, remove it. our functional automatically generate **btnBorderHColor** attribute. cross check it inside frontend.css.php and styling.js

File: src/blocks/cf7-designer/settings.js [hunk 14]
```diff
@@ -1158,48 +1114,13 @@ const Settings = ( props ) => {
...
-				} }
-				borderHoverColor={ {
-					value: buttonBorderHoverColor,
-					label: 'buttonBorderHoverColor',
-					title: __(
-						'Border Color',
-						'ultimate-addons-for-gutenberg'
-					),
-				} }
+				prefix={'button'}
```
Comment (imrantushar) [inline src/blocks/cf7-designer/settings.js:1119]: @supriyajjadhav if possible use **btn** prefix For cross copy/paste style

**Review by RajkiranBagal — APPROVED** [review]

File: includes/blocks/post-masonry/attributes.php (hunk 12 above)
Comment (imrantushar) [reply includes/blocks/post-masonry/attributes.php:104]: @supriyajjadhav are you remove `btnBorderHColor` attributes?

**Review by supriyajjadhav — APPROVED** [review]

**Pull request merged**